compiler/prelude/PrelNames.hs

```diff
@@ -2512,3 +2512,29 @@ pretendNameIsInScope n
   = any (n `hasKey`)
     [ liftedTypeKindTyConKey, tYPETyConKey
     , runtimeRepTyConKey, liftedRepDataConKey ]
+
+{-
+************************************************************************
+*                                                                      *
+    Daml extensions
+*                                                                      *
+************************************************************************
+-}
+
+dA_Internal_Desugar :: Module
+dA_Internal_Desugar = mkBaseModule (fsLit "DA.Internal.Desugar")
+
+hasInterfaceViewClassName, hasExerciseClassName, hasMethodClassName :: Name
+hasInterfaceViewClassName
+  = clsQual dA_Internal_Desugar (fsLit "HasInterfaceView") $ mkDamlClassOrMiscIdUnique 0
```

Comment:
in the daml repo this is defined in `DA.Internal.Interface`

Reply:
hm I realize this makes it harder to test these errors directly in this repo; even if we keep the module as `dA_Internal_Desugar` it's not going to pick up the definitions in `DA/Internal/Desugar.hs` because the unit id won't match. A temporary solution would be to move `DA/Internal/Desugar.hs` (while splitting it to resemble the structure in the daml repo) into `libraries/base` (reflecting that in `libraries/base/base.cabal`), but long-term I think we would want to replace `ghc-prim` & `base` entirely with a `daml-prim` in this repo - which raises the question of how to do that without duplicating `daml-prim` or adding a cyclic dependency between the repos.

that said, I think this depends on how much we want to test directly in this repo; currently it's a shorter feedback loop than going through the daml repo, but if that changes this becomes a moot point